src/ui/common/src/components/pages/workflows/index.tsx

```diff
@@ -218,21 +193,18 @@ const WorkflowsPage: React.FC<Props> = ({ user, Layout = DefaultLayout }) => {
     Metrics: (row) => {
       const workflowId = row.id;
 
-      const {
-        data: dagResults,
-        error: dagResultsError,
-        isLoading: dagResultsLoading,
-      } = useDagResultsGetQuery({
-        apiKey: user.apiKey,
-        workflowId: workflowId,
-      });
-
-      let latestDagResultId;
+      const { latestDagResult, dag: noRunDag } = useLatestDagResultOrDag(
+        user.apiKey,
+        workflowId
+      );
       let latestDagId;
-      if (!dagResultsLoading && !dagResultsError && dagResults.length > 0) {
-        const latestDagResult = getLatestDagResult(dagResults);
+      let latestDagResultId;
```

Comment:
I'd prefer using `const` if possible: https://stackoverflow.com/questions/41086633/why-most-of-the-time-should-i-use-const-instead-of-let-in-javascript#:~:text=Using%20const%20instead%20of%20let,somewhere%20else%20in%20the%20code.

Here we can use `const latestDagResultID = latestDagResult?.id` and `const dagId = `latestDagResult?.dag_id ?? dag?.id`

Reply:
See if this is still relevant after the suggested change in `useLatestDagResultOrDag`